longturn/freeciv21
# Monster refactor of shortcut management

Closes #1090 and more bugs I spotted while going through the code.
Closes #1088

See some detailed explanations in commit messages.

**Things to test:**
* Check that the defaults make sense
* Try all shortcuts, make sure they do something
* Change shortcuts and repeat the above, including assigning a mouse button to keyboard-like shortcuts (e.g. Reload Tileset) and a key sequence to mouse-like shortcuts (e.g. Scroll Map)
* Do it with different themes
* Check that the shortcuts are restored correctly on restart

Assigning two reviewers so you can split the work if you like...

*Note: the on-disk format of shortcuts has changed and there's no backward compatibility code*
